lms/djangoapps/hera/fragments.py

```diff
@@ -0,0 +1,200 @@
+                next_level = next_level if is_complete else lesson_pair[LEVEL]
+                first_sub, next_sub = get_pair(lesson_pair.get('block_id'), lesson_pair.get('next_id'), course_block_tree)
+                pair = []
+                # levels the same and at least one hasn't completed - return current pair
+                if lesson_pair[LEVEL] == next_level and (not first_sub.get('complete', True) or not next_sub.get('complete', True)):
+                    return {'children': [first_sub, next_sub]}
+            else: # means that student is here for the first time
+                next_level = get_next_level(earned)
+
+            for section in course_block_tree.get('children'):
+                for subsection in section.get('children', []):
+                    if next_level == subsection['unit_level'] and not subsection['complete']:
+                        subsection_vertical['children'].append(subsection)
+                        if len(subsection_vertical['children']) >= 2:
+                            break
+                        continue
+            to_set = {}
+            if subsection_vertical['children'] and len(subsection_vertical['children']) > 1:
+                to_set = {
+                        BLOCK_ID: subsection_vertical['children'][0]['block_id'],
+                        NEXT_ID: subsection_vertical['children'][1]['block_id'],
+                        LEVEL: next_level
+                    }
+            elif subsection_vertical['children']:
+                to_set = {
+                        BLOCK_ID: subsection_vertical['children'][0]['block_id'],
+                        LEVEL: next_level
+                }
+
+            if to_set:
+                c_selection_page().update({
+                    COURSE_KEY: course_id,
+                    USER: request.user.id},
+                    {'$set': to_set}
+                , upsert=True)
+
+            return subsection_vertical
+
+        context = {
+            'csrf': csrf(request)['csrf_token'],
+            'course': course_overview,
+            'due_date_display_format': course.due_date_display_format,
+            'blocks': course_block_tree
+        }
+
+        resume_block = get_resume_block(course_block_tree)
+        if not resume_block:
+            self.mark_first_unit_to_resume(course_block_tree)
+
+        xblock_display_names = self.create_xblock_id_and_name_dict(course_block_tree)
+        gated_content = self.get_content_milestones(request, course_key)
+
+        context['gated_content'] = gated_content
+        context['xblock_display_names'] = xblock_display_names
+        # earned = 0
+        try:
+            earned = round(earned/total*10, 1)
+        except ZeroDivisionError:
+            pass
+        context['subsection_vertical'] = unit_grade_level(earned, course_block_tree, subsection_active)
+
+        html = render_to_string('course_experience/course-lessons-outline-fragment.html', context)
```

Comment:
It worth moving this template to the hera app's templates.

Reply:
Moved